chat.go

```diff
@@ -128,13 +136,20 @@ func (api *Client) PostEphemeralContext(ctx context.Context, channelID, userID s
 }
 
 // UpdateMessage updates a message in a channel
-func (api *Client) UpdateMessage(channelID, timestamp, text string) (string, string, string, error) {
-	return api.UpdateMessageContext(context.Background(), channelID, timestamp, text)
+func (api *Client) UpdateMessage(channelID, timestamp, text string, params UpdateMessageParameters) (string, string, string, error) {
```

Comment:
Changes to the function are necessary to fully support the [chat.update](https://api.slack.com/methods/chat.update) API endpoint. 

Presently, the UpdateMessage function here only accepts updates to the text of a message itself. However, if a user wants to, for example, modify the attachments of an existing message, the present implementation does not allow this. My modifications allow for the full use of all the available options for `chat.update` defined in the slack API docs.

Please let me know if that clarifies things a bit for you on this, I'd be more than happy to further explain if needed.

Reply:
Ok, I see what you mean with SendMessage. 

That being said, I still feel that if you are going to implement the functionality of an endpoint, you should implement its full functionality as defined by the API provider. 

Also, is the change really breaking? If an argument isn't provided it will simply not add any extra information to the API request call, no? Thus any code that already uses this functionality should not experience any changes?

Reply:
>is the change really breaking

yup definitely breaking. if code will not compile its a breaking change. We even consider some subtler changes where the API is identical but behaviour changes to be breaking.

We try to limit these events where possible. see issues related to API breaking changes that havent been done yet. This is the very reason UpdateMessage,PostMessage,DeleteMessage still exist. because removing them would break API compatibility and we wouldn't even be having this discussion. =)

>I still feel that if you are going to implement the functionality of an endpoint, you should implement its full functionality as defined by the API provider.

Agreed, we differ on the definition of implemented. =) I'm claiming everything you are trying to accomplish can be (with possibly some minor extensions) using the existing SendMessage API.

Reply:
Alright, I'll remove the changes for the sake of the rest of this PR.

Reply:
However, please elaborate on your example so I can make the implementation in my project

Reply:
see https://github.com/nlopes/slack/pull/321#discussion_r194229460 =)